Installation issue: kokkos-tools

### Steps to reproduce the issue

Reproduce:
```
spack install kokkos-tools
```

<details><summary>Output of `spack spec`</summary>

<pre>
$ spack spec kokkos-tools
 -   kokkos-tools@develop~ipo~mpi~papi build_system=cmake build_type=Release generator=make arch=linux-rhel8-sapphirerapids
[+]      ^cmake@3.31.6~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release arch=linux-rhel8-sapphirerapids
[+]          ^curl@8.11.1~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl arch=linux-rhel8-sapphirerapids
[+]              ^nghttp2@1.65.0 build_system=autotools arch=linux-rhel8-sapphirerapids
[+]                  ^diffutils@3.10 build_system=autotools arch=linux-rhel8-sapphirerapids
[+]                      ^libiconv@1.18 build_system=autotools libs:=shared,static arch=linux-rhel8-sapphirerapids
[+]              ^openssl@3.4.1~docs+shared build_system=generic certs=mozilla arch=linux-rhel8-sapphirerapids
[+]                  ^ca-certificates-mozilla@2025-02-25 build_system=generic arch=linux-rhel8-sapphirerapids
[+]                  ^perl@5.40.0+cpanm+opcode+open+shared+threads build_system=generic arch=linux-rhel8-sapphirerapids
[+]                      ^berkeley-db@18.1.40+cxx~docs+stl build_system=autotools patches:=26090f4,b231fcc arch=linux-rhel8-sapphirerapids
[+]                      ^bzip2@1.0.8~debug~pic+shared build_system=generic arch=linux-rhel8-sapphirerapids
[+]                      ^gdbm@1.23 build_system=autotools arch=linux-rhel8-sapphirerapids
[+]                          ^readline@8.2 build_system=autotools patches:=1ea4349,24f587b,3d9885e,5911a5b,622ba38,6c8adf8,758e2ec,79572ee,a177edc,bbf97f1,c7b45ff,e0013d9,e065038 arch=linux-rhel8-sapphirerapids
[+]              ^pkgconf@2.3.0 build_system=autotools arch=linux-rhel8-sapphirerapids
[e]          ^gcc@12.3.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran,go' arch=linux-rhel8-sapphirerapids
[+]          ^gcc-runtime@12.3.0 build_system=generic arch=linux-rhel8-sapphirerapids
[+]          ^ncurses@6.5~symlinks+termlib abi=none build_system=autotools patches:=7a351bc arch=linux-rhel8-sapphirerapids
[+]          ^zlib-ng@2.2.4+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-rhel8-sapphirerapids
[+]      ^compiler-wrapper@1.0 build_system=generic arch=linux-rhel8-sapphirerapids
[e]      ^gcc@12.3.0~binutils+bootstrap~graphite~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++' arch=linux-rhel8-sapphirerapids
[+]      ^gcc-runtime@12.3.0 build_system=generic arch=linux-rhel8-sapphirerapids
[e]      ^glibc@2.28 build_system=autotools arch=linux-rhel8-sapphirerapids
[+]      ^gmake@4.4.1~guile build_system=generic arch=linux-rhel8-sapphirerapids
[+]      ^kokkos@4.6.00+aggressive_vectorization~cmake_lang~compiler_warnings+complex_align~cuda~debug~debug_bounds_check~debug_dualview_modify_check~deprecated_code+examples~hip_relocatable_device_code~hpx~hpx_async_dispatch~hwloc~ipo~memkind~numactl+openmp~openmptarget~pic~rocm+serial+shared~sycl~tests~threads~tuning~wrapper build_system=cmake build_type=Release cxxstd=17 generator=make intel_gpu_arch=none arch=linux-rhel8-sapphirerapids
</pre></details>

Pinging maintainers:
@jennfshr @rbberger @vlkale 

### Error message

<details><summary>variorum-connector error</summary>

<pre>
/home/prlavin/spack/opt/spack/linux-sapphirerapids/compiler-wrapper-1.0-7lym2hxetrzyrunlq4ao2scbvzcr563p/libexec/spack/cpp: line 401: SPACK_CC_LINKER_ARG: ERROR: LINKER ARG WAS NOT SET, MAYBE THE PACKAGE DOES NOT DEPEND ON CC?
"cpp includes s_filename" failed
Usage /usr/bin/dtrace [--help] [-h | -G] [-C [-I<Path>]] -s File.d [-o <File>]
make[2]: *** [profiling/systemtap-connector/CMakeFiles/kp_systemtap_connector.dir/build.make:77: profiling/systemtap-connector/probes.h] Error 1
make[2]: *** Waiting for unfinished jobs....
/tmp/prlavin/spack-stage/spack-stage-kokkos-tools-develop-z73v4oxqkv7h5yu6lcg4w7x3r6lavibx/spack-src/profiling/variorum-connector/variorum-connector.cpp: In function 'void KokkosTools::VariorumConnector::variorum_call()':
/tmp/prlavin/spack-stage/spack-stage-kokkos-tools-develop-z73v4oxqkv7h5yu6lcg4w7x3r6lavibx/spack-src/profiling/variorum-connector/variorum-connector.cpp:79:24: error: 'variorum_get_power_json' was not declared in this scope; did you mean 'variorum_get_node_power_json'?
   79 |   int variorum_error = variorum_get_power_json(&s);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
      |                        variorum_get_node_power_json
[ 45%] Linking CXX shared library libkp_hwm.so
cd /tmp/prlavin/spack-stage/spack-stage-kokkos-tools-develop-z73v4oxqkv7h5yu6lcg4w7x3r6lavibx/spack-build-z73v4ox/profiling/memory-hwm && /home/prlavin/spack/opt/spack/linux-sapphirerapids/cmake-3.31.6-x77iwkobjjnl4behj7gekzmce5yrolmw/bin/cmake -E cmake_link_script CMakeFiles/kp_hwm.dir/link.txt --verbose=1
[ 48%] Linking CXX shared library libkp_kernel_logger.so
cd /tmp/prlavin/spack-stage/spack-stage-kokkos-tools-develop-z73v4oxqkv7h5yu6lcg4w7x3r6lavibx/spack-build-z73v4ox/debugging/kernel-logger && /home/prlavin/spack/opt/spack/linux-sapphirerapids/cmake-3.31.6-x77iwkobjjnl4behj7gekzmce5yrolmw/bin/cmake -E cmake_link_script CMakeFiles/kp_kernel_logger.dir/link.txt --verbose=1
make[2]: *** [profiling/variorum-connector/CMakeFiles/kp_variorum_connector.dir/build.make:82: profiling/variorum-connector/CMakeFiles/kp_variorum_connector.dir/variorum-connector.cpp.o] Error 1
make[2]: Leaving directory '/ram/tmp/prlavin/spack-stage/spack-stage-kokkos-tools-develop-z73v4oxqkv7h5yu6lcg4w7x3r6lavibx/spack-build-z73v4ox'
make[1]: *** [CMakeFiles/Makefile2:819: profiling/variorum-connector/CMakeFiles/kp_variorum_connector.dir/all] Error 2
</pre></details>
<details><summary>dtrace error</summary>

<pre>
[ 69%] Built target kp_space_time_stack
Traceback (most recent call last):
  File "/usr/bin/dtrace", line 464, in <module>
    sys.exit(main())
  File "/usr/bin/dtrace", line 384, in main
    (ignore, fname) = mktemp_determ(["use_cpp", s_filename], suffix=".d")
  File "/usr/bin/dtrace", line 290, in mktemp_determ
    raise Exception("cannot create temporary file \""+fname+"\"")
Exception: cannot create temporary file ".dtrace-temp.5956373f.d"
</pre></details>

### Information on your system

$ spack debug report
* **Spack:** 1.0.0.dev0 (70f4eef020a71b8b5eed726e7304f45b04ca7537)
* **Python:** 3.6.8
* **Platform:** linux-rhel8-sapphirerapids

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/20299679/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/20299678/spack-build-out.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

Hi @vlkale The following tells me that an older version of Variorum is getting picked up in this build. 
Variorum API changed from `variorum_get_node_power_json` to `variorum_get_power_json` in v0.8. 
This has [already been updated](https://github.com/kokkos/kokkos-tools/blob/132108273f99891ebdf7f7d97fc639ccdfa7a379/profiling/variorum-connector/variorum-connector.cpp#L79) in the Kokkos connector last year by @masterleinad. 

Are you installing Variorum through spack or manually? 
If you're using spack to install Variorum -- that  package is old and being updated in PR spack/spack#50448. It is close to being merged so should be available relatively soon.
If installing manually, please make sure that the Variorum version is at 0.8 (or use the tip of dev).

```
/tmp/prlavin/spack-stage/spack-stage-kokkos-tools-develop-z73v4oxqkv7h5yu6lcg4w7x3r6lavibx/spack-src/profiling/variorum-connector/variorum-connector.cpp:79:24: error: 'variorum_get_power_json' was not declared in this scope; did you mean 'variorum_get_node_power_json'?
   79 |   int variorum_error = variorum_get_power_json(&s);
      |                        ^~~~~~~~~~~~~~~~~~~~~~~
      |                        variorum_get_node_power_json
```

--- a/repos/spack_repo/builtin/packages/kokkos_tools/package.py
+++ b/repos/spack_repo/builtin/packages/kokkos_tools/package.py
@@ -28,6 +28,12 @@ class KokkosTools(CMakePackage):
 
     def cmake_args(self):
         args = [
+            self.define("CMAKE_DISABLE_FIND_PACKAGE_Variorum", True),
+            self.define("KokkosTools_ENABLE_APEX", False),
+            self.define("KokkosTools_ENABLE_CALIPER", False),
+            self.define("KokkosTools_ENABLE_EXAMPLES", False),
+            self.define("KokkosTools_ENABLE_SINGLE", False),
+            self.define("KokkosTools_ENABLE_TESTS", False),
             self.define_from_variant("KokkosTools_ENABLE_MPI", "mpi"),
             self.define_from_variant("KokkosTools_ENABLE_PAPI", "papi"),
         ]

--- a/repos/spack_repo/builtin/packages/kokkos_tools/package.py
+++ b/repos/spack_repo/builtin/packages/kokkos_tools/package.py
@@ -28,9 +28,10 @@ class KokkosTools(CMakePackage):
 
     def cmake_args(self):
         args = [
-            self.define("CMAKE_DISABLE_FIND_PACKAGE_Variorum", True),
             self.define("KokkosTools_ENABLE_APEX", False),
             self.define("KokkosTools_ENABLE_CALIPER", False),
+            self.define("KokkosTools_ENABLE_SYSTEMTAP", False),
+            self.define("KokkosTools_ENABLE_VARIORUM", False),
             self.define("KokkosTools_ENABLE_EXAMPLES", False),
             self.define("KokkosTools_ENABLE_SINGLE", False),
             self.define("KokkosTools_ENABLE_TESTS", False),

--- a/repos/spack_repo/builtin/packages/kokkos_tools/package.py
+++ b/repos/spack_repo/builtin/packages/kokkos_tools/package.py
@@ -27,6 +27,9 @@ class KokkosTools(CMakePackage):
     depends_on("papi", when="+papi")
 
     def cmake_args(self):
+        # The plugins are intentionally disabled the time to properly introduce new variants
+        # with associated dependencies.
+        # Feel free to contribute.
         args = [
             self.define("KokkosTools_ENABLE_APEX", False),
             self.define("KokkosTools_ENABLE_CALIPER", False),

